### PR TITLE
feat: Compatible with replayResult interface logic

### PR DESCRIPTION
**Pull request by pangdayuan1**
Compatible with replayResult interface logic.

1. needMatch == true, maintain original logic, record and replay data through compareKey.
2. needMatch == false, there is no need to map the matching relationship, just compare directly

### CODE REVIEW

File: arex-schedule-web-api/src/main/java/com/arextest/schedule/client/ZstdJacksonMessageConverter.java [hunk 1]
```diff
@@ -41,6 +41,6 @@ protected Object readInternal(Class<?> clazz, HttpInputMessage inputMessage) thr
   @Override
   protected void writeInternal(Object o, HttpOutputMessage outputMessage) throws IOException,
       HttpMessageNotWritableException {
-    zstdJacksonSerializer.serializeTo(o, outputMessage.getBody());
+    outputMessage.getBody().write(zstdJacksonSerializer.serialize(o));
```
Comment (mr3) [inline arex-schedule-web-api/src/main/java/com/arextest/schedule/client/ZstdJacksonMessageConverter.java:44]: What is the purpose of the change codes here？

File: arex-schedule-web-api/src/main/java/com/arextest/schedule/client/HttpWepServiceApiClient.java [hunk 2]
```diff
@@ -237,6 +238,18 @@ public <TRequest, TResponse> TResponse retryJsonPost(String url, TRequest reques
     }
   }
 
+  public <TRequest, TResponse> TResponse retryJsonPost(String url, TRequest request,
```
Comment (mr3) [inline arex-schedule-web-api/src/main/java/com/arextest/schedule/client/HttpWepServiceApiClient.java:241]: `retryJsonPost`. maybe there's one more method: `retryZstdJsonPost`?

File: arex-schedule-web-api/src/main/java/com/arextest/schedule/comparer/impl/PrepareCompareItemBuilder.java [hunk 3]
```diff
@@ -44,6 +45,16 @@ CompareItem build(AREXMocker instance) {
...
   }
 
+  public CompareItem build(CompareReplayResult result, boolean recordCompareItem) {
+    String operationKey = result.getOperationName();
+    if (recordCompareItem) {
+      return new CompareItemImpl(operationKey, result.getBaseMsg(), null,
+          result.getRecordTime(), result.getCategoryType().isEntryPoint());
+    }
+    return new CompareItemImpl(operationKey, result.isSameMsg() ? result.getBaseMsg() : result.getTestMsg(),
+        null, result.getReplayTime(), result.getCategoryType().isEntryPoint());
```
Comment (mr3) [inline arex-schedule-web-api/src/main/java/com/arextest/schedule/comparer/impl/PrepareCompareItemBuilder.java:55]: Is a CompareItemBuilder needed here? put the build codes into the builder?

File: arex-schedule-web-api/src/main/java/com/arextest/schedule/client/ZstdJacksonMessageConverter.java (hunk 1 above)
Comment (pangdayuan1) [reply arex-schedule-web-api/src/main/java/com/arextest/schedule/client/ZstdJacksonMessageConverter.java:44]: outputMessage.body is OutputStream，i need ByteArrayOutputStream to echo the arex-storage project

File: arex-schedule-web-api/src/main/java/com/arextest/schedule/comparer/impl/PrepareCompareSourceRemoteLoader.java [hunk 4]
```diff
@@ -78,58 +79,103 @@ private QueryReplayResultResponseType remoteLoad(String replayId, String resultI
...
+    }
+
+    List<CategoryComparisonHolder> decodedListResult = new ArrayList<>(replayResults.size());
+    for (CompareRelationResult result : replayResults) {
+      MockCategoryType categoryType = result.getCategoryType();
+      if (categoryType == null || (!categoryType.isEntryPoint() && categoryType.isSkipComparison())) {
+        continue;
+      }
+
+      if (categoryType.isEntryPoint() && !categoryType.equals(MockCategoryType.Q_MESSAGE_CONSUMER) &&
```
Comment (binyu1005) [inline arex-schedule-web-api/src/main/java/com/arextest/schedule/comparer/impl/PrepareCompareSourceRemoteLoader.java:164]: 这里的判断是否可以优化？不要对MockCategoryType类型值进行硬编码判断

File: arex-schedule-web-api/src/main/java/com/arextest/schedule/comparer/CategoryComparisonHolder.java [hunk 5]
```diff
@@ -11,6 +12,18 @@
 public class CategoryComparisonHolder {
```
Comment (binyu1005) [inline arex-schedule-web-api/src/main/java/com/arextest/schedule/comparer/CategoryComparisonHolder.java:12]: 这个类型是否可以新增一个新的Holder类，而不是共用以前的类型，将来老逻辑下线删除起来也方便。

File: arex-schedule-web-api/src/main/java/com/arextest/schedule/service/PlanProduceService.java [hunk 6]
```diff
@@ -182,6 +182,12 @@ public ReplayPlan build(BuildReplayPlanRequest request, PlanContext planContext)
     List<ServiceInstance> serviceInstances = planContext.targetActiveInstance();
     replayPlan.setTargetHost(getIpAddress(serviceInstances));
     replayPlan.setTargetEnv(request.getTargetEnv());
+    ConfigurationService.ApplicationInstance instance =
+        configurationService.applicationInstance(appId, serviceInstances.get(0).getIp());
```
Comment (binyu1005) [inline arex-schedule-web-api/src/main/java/com/arextest/schedule/service/PlanProduceService.java:186]: 开源版本中 Instance中存的IP和实际用户填写的ip并不是同一个ip。Instance中的ip很可能是容器ip。这样的写法是否会有问题？

**Review by binyu1005 — DISMISSED** [review]

**Review by srotony — APPROVED** [review]

**Review by binyu1005 — APPROVED** [review]
